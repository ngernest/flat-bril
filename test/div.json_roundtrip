{
  "functions": [
    {
      "args": [],
      "instrs": [
        {
          "dest": "v0",
          "op": "const",
          "type": "int",
          "value": 9
        },
        {
          "dest": "v1",
          "op": "const",
          "type": "int",
          "value": -20
        },
        {
          "args": [
            "v0",
            "v1"
          ],
          "dest": "res",
          "funcs": [],
          "labels": [],
          "op": "div",
          "type": "int"
        },
        {
          "args": [
            "res"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
