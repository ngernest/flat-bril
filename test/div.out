@main
	op: const		dest: v0		type: int		value: 9	
	op: const		dest: v1		type: int		value: -20	
	op: div  		dest: res		type: int		args: ["v0", "v1"]	
	op: print		args: ["res"]	
